scene/gui/control.cpp

```diff
@@ -1383,6 +1383,7 @@ void Control::_set_size(const Size2 &p_size) {
 }
 
 void Control::set_size(const Size2 &p_size, bool p_keep_offsets) {
+	ERR_FAIL_COND(isnan(p_size.x) || isnan(p_size.y));
```

Comment:
Is this a relevant change?

Reply:
yes.. `temp_variable512.set_size(-NAN, -NAN)` crashes.

Reply:
```
func _ready() -> void:
	var temp_variable512 = GraphEdit.new()
	add_child(temp_variable512)
	temp_variable512.set_size(Vector2(-NAN, -NAN))
	temp_variable512.set_zoom(temp_variable512.get_zoom() * temp_variable512.get_zoom_step())
```
Yep It crashes...

Reply:
There is no exposed `_zoom_minus` or `_zoom_plus` in godot4... so i replace them with this `temp_variable512.set_zoom(temp_variable512.get_zoom() * temp_variable512.get_zoom_step())`

Reply:
Yes, I get that setting the size crashes, but it's not specific to GraphEdit, it's a method of Control. And the report doesn't even mention it. It mentions `set_begin` instead, which is also a method of Control. You can fix it in the same PR, I just find it kind of random to patch only that specific method of Control. I think either all relevant methods need to have these guards, or none.

Reply:
`Hmm.. may be we could do wrapper function to guard without touching the `Control` node?

Reply:
Well, my point was that we should add guards to Control methods, but it shouldn't be one arbitrary method. There are several sizing related methods that can be guarded.

Reply:
Oh.. I see I will look into it.